cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeHandler.java

```diff
@@ -143,15 +150,30 @@ public BodyConsumer writeMessages(HttpRequest request, HttpResponder responder,
     ProgramRunId programRunId = new ProgramRunId(appId,
                                                  ProgramType.valueOfCategoryName(programType, BadRequestException::new),
                                                  program, run);
-    requestValidator.validate(programRunId, request);
+    RunRecordStatus runRecordStatus = requestValidator.checkProgramRunStatus(programRunId, request);
 
     if (!allowedTopics.contains(topic)) {
       throw new UnauthorizedException("Access denied for topic " + topic);
     }
 
     TopicId topicId = NamespaceId.SYSTEM.topic(topic);
     if (topic.startsWith(logsTopicPrefix)) {
-      return new MessageBodyConsumer(topicId, logProcessor::process);
+      return new MessageBodyConsumer(topicId, logProcessor::process, runRecordStatus);
+    }
+
+    // after 10 mins add stopping state and timeout here
```

Comment:
the below lines is only for testing and will be removed before merging

Reply:
Usually we only keep the code changes in the PR that needs to be reviewed and merge. I think its safe to remove it from this PR